Dockerfile

```diff
@@ -0,0 +1,21 @@
+FROM openjdk:8
+
+# Install the OSS Review Toolkit
+ENV APPDIR=/opt/oss-review-toolkit
+COPY . "${APPDIR}"
+WORKDIR "${APPDIR}"
+RUN ./gradlew installDist
+
+# Install dependencies for NPM scanning
+RUN ["/bin/bash", "-c", "set -o pipefail && curl -sL https://deb.nodesource.com/setup_8.x | bash -"]
+RUN ["/bin/bash", "-c", "set -o pipefail && curl -sS https://dl.yarnpkg.com/debian/pubkey.gpg | apt-key add -"]
+RUN ["/bin/bash", "-c", "set -o pipefail && echo \"deb https://dl.yarnpkg.com/debian/ stable main\" | tee /etc/apt/sources.list.d/yarn.list"]
+RUN apt-get update && apt-get install -y apt-utils 
+RUN apt-get install -y build-essential python-pip nodejs yarn && pip install virtualenv && rm -rf /var/lib/apt/lists
```

Comment:
I wonder whether we should hard-code the tool versions we currently require by default. That's the way we currently do it in our internal `Dockerfile`:

    ...
    # install Node.js 8.x and NPM 5.3.0 (see: https://nodejs.org/en/download/package-manager/#debian-and-ubuntu-based-linux-distributions)
        curl -sL https://deb.nodesource.com/setup_8.x | bash - && \
        apt-get install -y nodejs && \
        npm install -g npm@5.5.1 && \
    # install Yarn (see: https://yarnpkg.com/lang/en/docs/install/#linux-tab)
        curl -sS https://dl.yarnpkg.com/debian/pubkey.gpg | apt-key add - && \
        echo "deb https://dl.yarnpkg.com/debian/ stable main" | tee /etc/apt/sources.list.d/yarn.list && \
        apt-get update && \
        apt-get install -y --no-install-recommends yarn=1.3.2-1 && \
    ...

Reply:
Publishing the image will lockdown the versions at the time you build the image but if you rebuilt the image then the versions would change. The trade-off is more management of the Docker image. Let me know what you'd like and I'll gladly make the change.

Reply:
I tend to prefer the fixed versions to not depend on a "random" version that was current at the time when the image was built, even if that adds more maintenance.

@mnonnenmacher, what's you opinion here?

Reply:
Also, I believe we should merge this `RUN` line with the one above, so `apt-get update` and `apt-get install` are always run together in the same layer. Otherwise, `apt update` may not be run if we updated only current line 14 [due to layer caching](https://docs.docker.com/engine/userguide/eng-image/dockerfile_best-practices/#apt-get).

Speaking of `apt-get`, these days I guess we should simply use `apt`.

Reply:
Good point, I'll make that change.

Reply:
@sschuberth I agree that we should use fixed versions of the tools. We only test specific versions and should guarantee that these tested versions are installed in the container.

Reply:
We also need to install these for the downloader: cvs, subversion, mercurial, git

Reply:
Correct. My thinking was to only do the obvious fixes for now, then merge, then align the committed with our internal Docker file, publish the container, then change CI to use the container to run tests.